### PR TITLE
[boost] update boost to 1.69. Fixes #5023 #4961

### LINKED ISSUE #5023: boost 1.69 is release please update

### PR DESCRIPTION
**Pull request by McDamon**
- Fixes #5023 #4961
- Remove boost-signal (rationale: deprecated and removed in 1.69)
- Remove boost-hana workaround by xiangfan-ms #4186 #3579 (rationale: 1.69 incorporates required fixes in hana master branch; builds fine in VS2017 15.9)

### CODE REVIEW

**Comment by huangqinjin** [comment]
New library `SafeNumerics` added in 1.69. Please add the port, thx @McDamon .

File: ports/boost-ptr-container/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Automatically generated by boost-vcpkg-helpers/generate-ports.ps1
 Source: boost-ptr-container
-Version: 1.68.0
+Version: 1.69.0
 Build-Depends: boost-array, boost-assert, boost-circular-buffer, boost-config, boost-core, boost-iterator, boost-mpl, boost-range, boost-serialization, boost-smart-ptr, boost-static-assert, boost-type-traits, boost-unordered, boost-utility, boost-vcpkg-helpers
```
Comment (janisozaur) [inline ports/boost-ptr-container/CONTROL]: It appears `boost-ptr-container` also depends on `boost-thread`, at least with `arm-windows` triplet in 1.68.

**Comment by janisozaur** [comment]
Also `boost-locale` depends on `boost-system` and `boost-thread`

**Comment by ras0219-msft** [comment]
Outstanding issues before we can merge this:

```
Processing arm-uwp                        313 vs 366
    boost-chrono                  **regression**                Fail vs Pass
    boost-regex                   **regression**                Fail vs Pass
    boost-signals                 **regression**                Fail vs Pass
Processing x64-osx                        524 vs 524
    libtorrent                    **regression**                Fail vs Pass
Processing x64-uwp                        402 vs 402
Processing x64-windows                    796 vs 800
    cartographer                  **regression**                Fail vs Pass
    hpx                           **regression**                Fail vs Pass
    libtorrent                    **regression**                Fail vs Pass
    wt                            **regression**                Fail vs Pass
Processing x64-windows-static             712 vs 714
    hpx                           **regression**                Fail vs Pass
    libtorrent                    **regression**                Fail vs Pass
Processing x86-windows                    779 vs 783
    cartographer                  **regression**                Fail vs Pass
    hpx                           **regression**                Fail vs Pass
    libtorrent                    **regression**                Fail vs Pass
    wt                            **regression**                Fail vs Pass
```

**Comment by janisozaur** [comment]
I filed https://github.com/boostorg/type_traits/issues/105 which affects compilation of `boost-regex` and some more packages and it was fixed in https://github.com/boostorg/type_traits/commit/5e7844bea12485c89e14cd08f180aeb401b054b5 so you may want to take that patch to fix arm builds. Subsequently, MSVC/ARM builds were added to `type_traits`' AppVeyor array.

**Comment by jasjuang** [comment]
Looking forward to see the problem getting resolved and merged!

**Comment by janisozaur** [comment]
@ras0219-msft I noticed your latest commit and that makes me wonder: I was recently trying to build a project using `boost-iostreams` and ran into lining errors regarding zlib and bzip2, like

>
```
error LNK2001: unresolved external symbol "protected: __thiscall boost::iostreams::detail::bzip2_base::~bzip2_base(void)" (??1bzip2_base@detail@iostreams@boost@@IAE@XZ)
error LNK2001: unresolved external symbol "protected: __thiscall boost::iostreams::detail::bzip2_base::bzip2_base(struct boost::iostreams::bzip2_params const &)" (??0bzip2_base@detail@iostreams@boost@@IAE@ABUbzip2_params@23@@Z)
```
(These errors are not mine, but look close enough)

I will try building when time allows to test with your latest change

**Comment by McDamon** [comment]
> New library `SafeNumerics` added in 1.69. Please add the port, thx @McDamon .

Added

**Pull request merged**